controllers/aggregated_clusteroperator_controller.go

```diff
@@ -0,0 +1,148 @@
+	utilerror "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/discovery"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logr "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	platformv1alpha1 "github.com/openshift/platform-operators/api/v1alpha1"
+	aggregatedco "github.com/openshift/platform-operators/internal/aggregated-co"
+	"github.com/openshift/platform-operators/internal/util"
+)
+
+type AggregatedClusterOperatorReconciler struct {
+	client.Client
+	DiscoveryClient discovery.DiscoveryInterface
+	Configv1Client  configv1client.ConfigV1Interface
+}
+
+const aggregateCOName = "platform-operators-aggregated"
+
+//+kubebuilder:rbac:groups=platform.openshift.io,resources=platformoperators,verbs=get;list;watch
+//+kubebuilder:rbac:groups=platform.openshift.io,resources=platformoperators/status,verbs=get
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusteroperators,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (a *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logr.FromContext(ctx)
+	log.Info("reconciling request", "req", req.NamespacedName)
+	defer log.Info("finished reconciling request", "req", req.NamespacedName)
+
+	// Create a CO Builder to build the CO status
+	coBuilder := aggregatedco.NewBuilder()
+	// Create a CO Writer to write to the CO status
+	coWriter := aggregatedco.NewWriter(a.Configv1Client)
+
+	aggregatedCO := &openshiftconfigv1.ClusterOperator{}
+	if err := a.Get(ctx, req.NamespacedName, aggregatedCO); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	defer func() {
+		if err := coWriter.UpdateStatus(aggregatedCO, coBuilder.GetStatus()); err != nil {
+			log.Error(err, "error updating CO status")
+		}
+	}()
+
+	// Set the default CO status conditions: Progressing True, Degraded False, Available False
+	coBuilder.WithProgressing(openshiftconfigv1.ConditionTrue, "")
+	coBuilder.WithDegraded(openshiftconfigv1.ConditionFalse)
+	coBuilder.WithAvailable(openshiftconfigv1.ConditionFalse, "", "")
+
+	poList := &platformv1alpha1.PlatformOperatorList{}
+	if err := a.List(ctx, poList); err != nil {
+		log.Error(err, "error listing platformoperators")
+		return ctrl.Result{}, err
+	}
+
+	if len(poList.Items) == 0 {
+		// No POs on cluster, everything is fine
+		coBuilder.WithAvailable(openshiftconfigv1.ConditionTrue, "", "No POs Found")
```

Comment:
nit: It looks like WithAvailable is expecting a status condition reason here which are typically in the form of camel-cased, one-word phrases.